### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by kimitaWanjohi**
In this PR we did the following:
- Modularized the code in files 
- Removed repetitions.
- Refactored classes to ensure that each class only has one responsibility

### CODE REVIEW

File: classes/list_class.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        puts 'No rentals found for this person.'
+      else
+        puts "Rentals for #{person.name}:"
+        person.rentals.each do |rental|
+          puts "Date: #{rental.date}, Book: #{rental.book.title}, Author: #{rental.book.author}"
+        end
+      end
+    else
+      puts "Person with ID #{id} not found."
+    end
```
Comment (mckent05) [inline classes/list_class.rb:39]: - You have done well refactoring so far, but I have an error when I try to list the rentals for ID, please fix this 🙏 
gcgchchgc

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @kimitaWanjohi, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good refactoring for code 👏 👏 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/list_class.rb (hunk 1 above)
Comment (Kgomotso196) [reply classes/list_class.rb:39]: @mckent05 , thank you for your constructive feedback, working on it.

Comment (Abenezer-Tilahun) [reply classes/list_class.rb:39]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @kimitaWanjohi 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**